packages/gatsby-source-wordpress/src/utils/run-steps.ts

```diff
@@ -108,12 +109,12 @@ const findApiName = (initialApiNameString: string): string => {
   )
 }
 
-const runApiSteps =
-  (steps: Array<Step>, apiName: string) =>
-  async (
-    helpers: GatsbyNodeApiHelpers,
-    pluginOptions: IPluginOptions
-  ): Promise<void> =>
-    runSteps(steps, helpers, pluginOptions, apiName)
+const runApiSteps = (steps: Array<Step>, apiName: string): IGatsbyApiHook =>
+  wrapApiHook(
+    async (
+      helpers: GatsbyNodeApiHelpers,
+      pluginOptions: IPluginOptions
+    ): Promise<void> => runSteps(steps, helpers, pluginOptions, apiName)
+  )
```

Comment:
We wrap the function in `wrapApiHook`, which adds the handling for AsyncLocalStorage